pkg/cli/admin/prune/imageprune/prune_test.go

```diff
@@ -978,22 +997,28 @@ func TestImagePruning(t *testing.T) {
 			expectedImageDeletions: []string{"sha256:0000000000000000000000000000000000000000000000000000000000000004", "sha256:0000000000000000000000000000000000000000000000000000000000000005"},
 			expectedStreamUpdates:  []string{"foo/bar|sha256:0000000000000000000000000000000000000000000000000000000000000004"},
 			expectedLayerLinkDeletions: []string{
-				registryURL + "|foo/bar|" + imagetest.Config2,
-				registryURL + "|foo/bar|layer5",
-				registryURL + "|foo/bar|layer6",
-				registryURL + "|foo/bar|layer7",
-				registryURL + "|foo/bar|layer8",
+				"foo/bar|" + imagetest.Config2,
+				"foo/bar|layer5",
+				"foo/bar|layer6",
+				"foo/bar|layer7",
+				"foo/bar|layer8",
 			},
-			expectedManifestLinkDeletions: []string{registryURL + "|foo/bar|sha256:0000000000000000000000000000000000000000000000000000000000000004"},
+			expectedManifestLinkDeletions: []string{"foo/bar|sha256:0000000000000000000000000000000000000000000000000000000000000004"},
 			expectedBlobDeletions: []string{
-				registryURL + "|sha256:0000000000000000000000000000000000000000000000000000000000000004",
-				registryURL + "|sha256:0000000000000000000000000000000000000000000000000000000000000005",
-				registryURL + "|layer7",
-				registryURL + "|layer8",
-				registryURL + "|layer9",
-				registryURL + "|layerX",
+				"sha256:0000000000000000000000000000000000000000000000000000000000000004",
+				"sha256:0000000000000000000000000000000000000000000000000000000000000005",
+				imagetest.Config2,
+				"layer5",
+				"layer6",
```

Comment:
Could you explain why do you think they should be also removed? As far as I understand everything here should be removed, **EVERYTHING** as there is no pod/deployment/rs/dc making reference to any of those. Trying to understand the pruner is hard.

Reply:
@ricardomaraschini the pruner by default keeps 3 latest revisions (it can be overrode by `--keep-tag-revisions`) so that it doesn't delete tags and their latest images.